File: api/src/main/java/com/ruby/rubia_server/core/dto/MessageTemplateRevisionDTO.java
package com.ruby.rubia_server.core.dto;

import com.ruby.rubia_server.core.enums.RevisionType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MessageTemplateRevisionDTO {

    private UUID id;
    private UUID templateId;
    private String templateName;
    private Integer revisionNumber;
    private String content;
    private UUID editedByUserId;
    private String editedByUserName;
    private RevisionType revisionType;
    private LocalDateTime revisionTimestamp;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    
    // Campos para metadados de IA
    private UUID aiAgentId;
    private String aiAgentName;
    private String aiEnhancementType;
    private Integer aiTokensUsed;
    private Integer aiCreditsConsumed;
    private String aiModelUsed;
    private String aiExplanation;
}

File: api-dotnet/Rubia.Server/DTOs/MessageTemplateRevisionDto.cs
using Rubia.Server.Enums;
using System.ComponentModel.DataAnnotations;

namespace Rubia.Server.DTOs;

public class MessageTemplateRevisionDto
{
    public Guid Id { get; set; }
    public Guid TemplateId { get; set; }
    public string? TemplateName { get; set; }
    public int RevisionNumber { get; set; }
    public string Content { get; set; } = string.Empty;
    public Guid? EditedByUserId { get; set; }
    public string? EditedByUserName { get; set; }
    public RevisionType RevisionType { get; set; }
    public DateTime RevisionTimestamp { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // AI metadata fields
    public Guid? AiAgentId { get; set; }
    public string? AiAgentName { get; set; }
    public string? AiEnhancementType { get; set; }
    public int? AiTokensUsed { get; set; }
    public int? AiCreditsConsumed { get; set; }
    public string? AiModelUsed { get; set; }
    public string? AiExplanation { get; set; }
}

public class CreateMessageTemplateRevisionDto
{
    [Required]
    public Guid TemplateId { get; set; }

    [Required]
    [MaxLength(10000)]
    public string Content { get; set; } = string.Empty;

    [Required]
    public RevisionType RevisionType { get; set; }

    public Guid? EditedByUserId { get; set; }

    // AI metadata
    public Guid? AiAgentId { get; set; }
    public string? AiEnhancementType { get; set; }
    public int? AiTokensUsed { get; set; }
    public int? AiCreditsConsumed { get; set; }
    public string? AiModelUsed { get; set; }
    public string? AiExplanation { get; set; }
}

public class UpdateMessageTemplateRevisionDto
{
    [MaxLength(10000)]
    public string? Content { get; set; }

    public RevisionType? RevisionType { get; set; }
    public string? AiEnhancementType { get; set; }
    public int? AiTokensUsed { get; set; }
    public int? AiCreditsConsumed { get; set; }
    public string? AiModelUsed { get; set; }
    public string? AiExplanation { get; set; }
}